rust/sqlite/src/db.rs

```diff
@@ -154,7 +154,12 @@ impl SqliteDb {
             }
         }
 
-        let unapplied = source_migrations[applied_migrations.len()..].to_vec();
+        let unapplied = if applied_migrations.len() < source_migrations.len() {
+            source_migrations[applied_migrations.len()..].to_vec()
+        } else {
+            Vec::<Migration>::new()
+        };
+
```

Comment:
Adding a new migration surfaced a bug in `test_cross_version_persist.py` where when we moved to an older version with persisted data, there were more applied migrations than source migrations (leading to slice out of bound error).